lib/config/database.go

```diff
@@ -0,0 +1,274 @@
+  # # For more information: https://goteleport.com/docs/database-access/guides/postgres-redshift/
+  # - name: redshift
+  #   description: AWS Redshift cluster configuration example.
+  #   # Supported protocols for Redshift: "postgres" or "mysql"
+  #   protocol: postgres
+  #   # Database connection endpoint. Must be reachable from Database service.
+  #   uri: redshift-cluster-example-1.abcdefghijklmnop.us-west-1.redshift.amazonaws.com:5439
+  #   # AWS specific configuration.
+  #   aws:
+  #     # Region the database is deployed in.
+  #     region: us-west-1
+  #     # Redshift specific configuration.
+  #     redshift:
+  #       # Redshift Cluster ID.
+  #       cluster_id: redshift-cluster-example-1
+  # # Self-hosted static configuration.
+  # - name: self-hosted
+  #   description: Self-hosted database configuration.
+  #   # Supported protocols for self-hosted: {{ .SupportedDatabaseProtocols }}.
+  #   protocol: postgres
+  #   # Database connection endpoint. Must be reachable from Database service.
+  #   uri: database.example.com:5432
+  {{- end }}
+auth_service:
+  enabled: "no"
+ssh_service:
+  enabled: "no"
+proxy_service:
+  enabled: "no"`))
+
+// DatabaseSampleFlags specifies configuration parameters for a database agent.
+type DatabaseSampleFlags struct {
+	// StaticDatabaseName static database name provided by the user.
+	StaticDatabaseName string
+	// StaticDatabaseProtocol static databse protocol provided by the user.
+	StaticDatabaseProtocol string
+	// StaticDatabaseURI static database URI provided by the user.
+	StaticDatabaseURI string
+	// StaticDatabaseStaticLabels list of database static labels provided by
+	// the user.
+	StaticDatabaseStaticLabels map[string]string
+	// StaticDatabaseDynamicLabels list of database dynamic labels provided by
+	// the user.
+	StaticDatabaseDynamicLabels services.CommandLabels
+	// StaticDatabaseRawLabels "raw" list of database labels provided by the
+	// user.
+	StaticDatabaseRawLabels string
+	// NodeName `nodename` configuration.
+	NodeName string
+	// DataDir `data_dir` configuration.
+	DataDir string
+	// ProxyServerAddr is a list of addresses of the auth servers placed on
+	// the configuration.
+	AuthServersAddr []string
+	// AuthToken auth server token.
+	AuthToken string
+	// CAPins are the SKPI hashes of the CAs used to verify the Auth Server.
+	CAPins []string
+	// RDSDiscoveryRegions list of regions the RDS auto-discovery is
+	// configured.
+	RDSDiscoveryRegions []string
+	// RedshiftDiscoveryRegions list of regions the Redshift auto-discovery is
+	// configured.
+	RedshiftDiscoveryRegions []string
+	// SupportedDatabaseProtocols list of database protocols supported.
+	SupportedDatabaseProtocols string
+}
+
+// CheckAndSetDefaults checks and sets default values for the flags.
+func (f *DatabaseSampleFlags) CheckAndSetDefaults() error {
+	conf := service.MakeDefaultConfig()
+	f.SupportedDatabaseProtocols = strings.Join(defaults.DatabaseProtocols, ", ")
```

Comment:
Seems like you're already using template Funcs so how about getting rid of this `SupportedDatabaseProtocols` field and instead defining a StringsJoin template func, passing defaults.DatabaseProtocols to the template and joining them there?

```go
template.FuncMap{"StringsJoin": strings.Join}

{{ StringsJoin .DatabaseProtocols ", " }}
```